modules/swagger-codegen/src/main/resources/csharp/ApiClient.mustache

```diff
@@ -145,7 +145,7 @@ namespace {{packageName}}.Client
         /// <param name="contentType">Content Type of the request</param>
         /// <returns>Object</returns>
         public Object CallApi(
-            String path, RestSharp.Method method, Dictionary<String, String> queryParams, Object postBody,
+            String path, RestSharp.Method method, List<KeyValuePair<String, String>> queryParams, Object postBody,
```

Comment:
I would think this needs to be a method overload. There's no direct conversion between `Dictionary<String, String>` and `List<KeyValuePair<String, String>>`, so newly generated clients would break existing callers.

Also, `CallApiAsync` needs to have the same reflected change.

The type `List<KeyValuePair<String, String>>` will probably also be confusing to some developers. Most people see `KeyValuePair<T, K>` and think "this belongs in a dictionary". I would think `Dictionary<String, List<String>>` would be more appropriate. A `List<Tuple<String,String>>` could also reduce confusion, but it's not exempt from the same issues as `List<KeyValuePair<String, String>>`.

The RFC for URIs (http://tools.ietf.org/html/rfc3986#section-6.2.2.1) claims that scheme and host are case insensitive, and everything else is case sensitive. This means the following won't result in multi values for the same key as some might expect:

```
var list = new List<KeyValuePair<String, String>>();
list.Add(new KeyValuePair("key", "value1"));
list.Add(new KeyValuePair("KEY", "value2"));
```

Obviously, you'd have the same problem with casing in a dictionary. But you'd be reducing the footprint for developer error:

```
var d = new Dictionary<String,List<String>>();
d["key"] = new List<String> { "value1", "value2" };
```